processor/memorylimiterprocessor/README.md

```diff
@@ -12,61 +12,72 @@
 [contrib]: https://github.com/open-telemetry/opentelemetry-collector-releases/tree/main/distributions/otelcol-contrib
 <!-- end autogenerated section -->
 
+## Overview
+
 The memory limiter processor is used to prevent out of memory situations on
 the collector. Given that the amount and type of data the collector processes is
-environment specific and resource utilization of the collector is also dependent
+environment-specific and resource utilization of the collector is also dependent
 on the configured processors, it is important to put checks in place regarding
 memory usage.
- 
-The memory_limiter processor allows to perform periodic checks of memory
-usage if it exceeds defined limits will begin refusing data and forcing GC to reduce
-memory consumption.
 
-The memory_limiter uses soft and hard memory limits. Hard limit is always above or equal
-the soft limit.
+## Functionality
+
+The memory limiter processor performs periodic checks of memory
+usage and will begin refusing data and forcing GC to reduce
+memory consumption when defined limits have been exceeded.
+
+The processor uses soft and hard memory limits. The hard limit is defined via the
+`limit_mib` configuration option, and is always above or equal
+to the soft limit. The difference between the soft limit and hard limit is defined via
+the `spike_limit_mib` configuration option.
 
-When the memory usage exceeds the soft limit the processor will enter the memory limited
-mode and will start refusing the data by returning errors to the preceding component
+The processor will enter memory limited mode and will start refusing the data when
+memory usage exceeds the soft limit. This is done by returning errors to the preceding component
 in the pipeline that made the ConsumeLogs/Trace/Metrics function call.
-The preceding component should be normally a receiver.
 
 In memory limited mode the error returned by ConsumeLogs/Trace/Metrics function is a
 non-permanent error. When receivers see this error they are expected to retry sending
-the same data. The receivers may also apply a backpressure to their data sources
-in order to slow down the inflow of data into the Collector and allow the memory usage
-to go below the limits.
+the same data. The receivers may also apply backpressure to their own data sources
+in order to slow the inflow of data into the Collector, and to allow memory usage
+to go below the set limits.
 
->Warning: if the component preceding the memory limiter in the pipeline does not correctly
-retry and send the data again after ConsumeLogs/Trace/Metrics functions return then that
-data will be permanently lost. We consider such components incorrectly implemented.
+>Warning: Data will be permanently lost if the component preceding the memory limiter
+> in the telemetry pipeline does not correctly retry sending data after the
+> ConsumeLogs/Trace/Metrics function returns.
```

Comment:
I dont expect users of the memory limiter to understand what these functions mean.  Can we reword this to something like "does not correctly retry sending data after the memory limiter processor refuses data."

Reply:
Good call, describing functionality is definitely easier to understand than referencing function names. 👍